docs/code-search/code-navigation/private-maven-repository-configuration-scip-java.mdx

```diff
@@ -0,0 +1,85 @@
+### Configuring Maven to Use a Private Repository for SCIP-JAVA
+
+#### Overview
+In Sourcegraph, configuring SCIP-JAVA to use a private Maven repository, such as Nexus or Artifactory, is essential when dependencies should not be retrieved from public repositories. This guide covers steps to set up Maven for SCIP-JAVA indexing with a private repository, ensuring secure and consistent dependency resolution.
+
+#### Step 1: Testing the Configuration on a Single Repository
+
+To test and validate the Maven configuration, start by modifying a single repository’s auto-indexing settings to include a custom `settings.xml` file.
+
+1. **Add Custom Index Job Configuration**:
+   - Access the repository’s index settings in Sourcegraph and open the “Raw” configuration panel.
+   - Insert the following configuration:
+
+     ```json
+     {
+       "steps": [],
+       "local_steps": [
+         "mkdir -p ~/.m2",
+         "echo '<settings xmlns=\"http://maven.apache.org/SETTINGS/1.0.0\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd\"> <servers> <server> <id>repo</id> <username>$ARTIFACTORY_USER</username> <password>$ARTIFACTORY_PASSWORD</password> </server> </servers> </settings>' > ~/.m2/settings.xml"
+       ],
+       "root": "",
+       "indexer": "sourcegraph/scip-java:latest-snapshot",
+       "indexer_args": [
+         "scip-java",
+         "index",
+         "--build-tool=auto"
+       ],
+       "outfile": "index.scip",
+       "requestedEnvVars": [
+         "ARTIFACTORY_USER",
+         "ARTIFACTORY_PASSWORD"
+       ]
+     }
+     ```
+
+2. **Trigger Indexing**:
+   - After configuring the repository, navigate to the "Precise Indexes" tab and click "Enqueue" to start the indexing process.
+   - Ensure that the environment variables `$ARTIFACTORY_USER` and `$ARTIFACTORY_PASSWORD` are created as Executor Secrets.
```

Comment:
Worth adding a link here: https://sourcegraph.com/docs/admin/executors/executor_secrets

Also the executor secrets step should be a separate step in the list, before `trigger indexing`